### PR TITLE
postinst cleanups + remove root_mode

### PR DESCRIPTION
**Pull request by orndorffgrant**
## Test Steps

LMK if you disagree with the postinst changes.
LMK if you disagree with the new util function for is_root (it's intended to be a bit more readable)
Integration tests should still pass.
Manually run the Pro fips tests.
Manually run the new notice code in postinst to make sure it works as expected.

## Checklist

 - [x] I have updated or added any unit tests accordingly
 - [x] I have updated or added any integration tests accordingly
 - [x] I have updated or added any documentation accordingly (these are all "under-the-hood" changes - no docs necessary)

## Does this PR require extra reviews?

 - [ ] Yes
 - [x] No (? unless you think we need extra reviews because of the changes to postinst)

### CODE REVIEW

File: uaclient/entitlements/tests/test_repo.py [hunk 1]
```diff
@@ -433,7 +433,7 @@ def test_enable_can_exit_on_pre_or_post_disable_messaging_hooks(
     @pytest.mark.parametrize("should_reboot", (False, True))
     @pytest.mark.parametrize("with_pre_install_msg", (False, True))
     @pytest.mark.parametrize("packages", (["a"], [], None))
-    @mock.patch("os.getuid", return_value=0)
+    @mock.patch("uaclient.util.we_are_currently_root", return_value=True)
```
[the remark this answers is not in the document]
Comment (orndorffgrant) [reply uaclient/entitlements/tests/test_repo.py:436]: Ah yes this is something I wanted to bring up. I agree, but I wanted to see if anyone disagreed with the autouse mock before going down that path. Let's just sync to confirm after standup tomorrow

[the remark this answers is not in the document]
Comment (orndorffgrant) [reply uaclient/entitlements/tests/test_repo.py]: Done!

File: uaclient/files/notices.py [hunk 2]
```diff
@@ -160,27 +160,48 @@ def list(self) -> List[Tuple[str, str]]:
...
+                        notice = next(
+                            iter(
+                                [
+                                    n
+                                    for n in Notice
+                                    if n.order_id == order_id
+                                    and n.label == label
+                                ]
+                            ),
+                        )
```
Comment (lucasmoura) [inline uaclient/files/notices.py:193]: This is just a style suggestion, so feel free to disagree here, but we can't we just loop the notices and raise an exception if we can find any that matches the empty notice name ?

File: features/ubuntu_pro_fips.feature [hunk 3]
```diff
@@ -527,6 +527,7 @@ Feature: Command behaviour when auto-attached in an ubuntu PRO fips image
         """
         1
         """
+        When I run `systemctl daemon-reload` with sudo
```
Comment (lucasmoura) [inline features/ubuntu_pro_fips.feature:530]: I know this is unrelated, but those pro-fips tests are failing due to the extra check for the `esm-infra-security` when running apt-cache policy on the tested package. Maybe it is worth an extra commit just to address that and fix those tests as well

**Review by lucasmoura — APPROVED** [review]

File: uaclient/files/notices.py (hunk 2 above)
Comment (orndorffgrant) [reply uaclient/files/notices.py:193]: I agree that will be more readable, I'll add it before merging

File: features/ubuntu_pro_fips.feature (hunk 3 above)
Comment (orndorffgrant) [reply features/ubuntu_pro_fips.feature:530]: Ah darn that wasn't happening when I was editing this. I'll see if I can fix it and tack on an extra commit here before merging

**Pull request merged**